engine/crates/runtime-local/src/entity_cache/memory.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    sync::Mutex,
+    time::Instant,
+};
+
+use futures_util::future::BoxFuture;
+
+#[derive(Default)]
+pub struct InMemoryEntityCache {
+    inner: Mutex<HashMap<String, CacheValue>>,
+}
+
+struct CacheValue {
+    data: Vec<u8>,
+    expires_at: Instant,
+}
+
+impl InMemoryEntityCache {
+    async fn get(&self, name: &str) -> anyhow::Result<Option<Vec<u8>>> {
+        let mut lock = self.inner.lock().unwrap();
```

Comment:
Should this be expect?

Reply:
It shouldn't even be in this PR, accidentally hit ready for review before I'd done a rebase, please ignore.